indico/modules/events/fields.py

```diff
@@ -121,72 +77,90 @@ class PersonLinkListFieldBase(EventPersonListField):
         if not person_link:
             person_link = self.person_link_cls(person=person)
-        person_link.populate_from_dict(person_data)
+        person_link.populate_from_dict(PersonLinkSchema(
+            only=('first_name', 'last_name', 'affiliation', 'address', 'phone', 'display_order')).load(data))
         email = data.get('email', '').lower()
+        if not email:
+            raise UserValueError(_('A valid email address is required'))
         if email != person_link.email:
             if not self.event or not self.event.persons.filter_by(email=email).first():
                 person_link.person.email = email
                 person_link.person.user = get_user_by_email(email)
                 if inspect(person).persistent:
                     signals.event.person_updated.send(person_link.person)
             else:
-                raise UserValueError(_('There is already a person with the email {}').format(email))
+                raise UserValueError(_('There is already a person with the email {email}').format(email=email))
         return person_link
 
-    def _serialize_principal(self, principal):
-        if not isinstance(principal, PersonLinkBase):
-            return super()._serialize_principal(principal)
-        if principal.id is None:
-            return super()._serialize_principal(principal.person)
-        else:
-            return self._serialize_person_link(principal)
+    def _serialize_person_link(self, principal):
+        raise NotImplementedError
 
-    def _serialize_person_link(self, principal, extra_data=None):
+    def _convert_data(self, data):
         raise NotImplementedError
 
     def _value(self):
         return [self._serialize_person_link(person_link) for person_link in self.data] if self.data else []
 
+    def process_formdata(self, valuelist):
+        if valuelist:
+            self.data = json.loads(valuelist[0])
+            try:
+                self.data = self._convert_data(self.data)
+            except ValueError:
+                self.data = []
+                raise
+
 
 class EventPersonLinkListField(PersonLinkListFieldBase):
     """A field to manage event's chairpersons."""
 
     person_link_cls = EventPersonLink
     linked_object_attr = 'event'
-    widget = JinjaWidget('events/forms/event_person_link_widget.html')
+    widget = JinjaWidget('forms/person_link_widget.html')
+
+    @property
+    def roles(self):
+        return [{'name': 'submitter', 'label': _('Submitter'), 'icon': 'paperclip',
+                 'default': self.default_is_submitter}]
 
     def __init__(self, *args, **kwargs):
-        self.allow_submitters = True
         self.default_is_submitter = kwargs.pop('default_is_submitter', True)
+        self.empty_message = _('There are no chairpersons')
+        event_type = kwargs.pop('event_type', EventType.conference)
         super().__init__(*args, **kwargs)
+        if event_type or (self.object.event and self.object.event.type_) == EventType.lecture:
```

Comment:
🤯  

But I don't think `event_type` is ever falsy... And if I understand it correctly, you always set either an event type via kwarg or have an event... so you could probably simplify this to something like this:

```python
event_type = self.object.event.type_ if self.object and self.object.event else kwargs.pop('event_type')
```

(can we ever have `self.object` without an event?)

Reply:
I had the feeling this line was problematic, it became confusing because we can't access `self.object` before the `super` as it doesn't exist until there, plus `event_type` needs to be "pop" before.

> (can we ever have self.object without an event?)

In the `LectureCreationForm`

Reply:
what's the object there? the category?

Reply:
The object is `self.object = getattr(kwargs['_form'], self.linked_object_attr, None)` where `linked_object_attr` is deemed to be an `event` or `contribution`, etc. There is none as the lecture is still being created.

Reply:
My latest changes should make it more clear.

Reply:
so `self.object` is already none - which is what i was asking about in "can we ever have self.object without an event?"

Reply:
Ah, yes - I don't recall exactly, I believe not with the current changes.